test/tests/interfacekernels/1d_interface/tests

```diff
@@ -1,45 +1,55 @@
 [Tests]
-  [./interface_diffusion]
+  [interface_diffusion]
     type = 'Exodiff'
     input = 'coupled_value_coupled_flux.i'
     exodiff = 'coupled_value_coupled_flux_out.e'
     requirement = "Interface kernels shall provide integrated conditions between subdomains,"
                   " and shall work with boundary restricted materials with stateful properties."
     issues = '#11258 #869'
     design = 'InterfaceKernels/index.md'
-  [../]
-  [./interface_diffusion_penalty]
-    type = 'Exodiff'
-    input = 'coupled_value_coupled_flux.i'
-    cli_args = "InterfaceKernels/active='penalty_interface' BCs/active='left right' Outputs/file_base=coupled_value_coupled_flux_penalty_out"
-    exodiff = 'coupled_value_coupled_flux_penalty_out.e'
-    issues = '#11765'
+  []
+  [interface_diffusion_penalty]
     requirement = "The interface diffusion penalty method should reproduce the analytic solution"
-    prereq = 'interface_diffusion'
     design = 'InterfaceKernels/index.md'
-  [../]
-  [./interface_diffusion_penalty_with_jump_material]
+    issues = '#11765'
+    [hand-coded]
+      type = 'Exodiff'
+      input = 'coupled_value_coupled_flux.i'
+      cli_args = "InterfaceKernels/active='penalty_interface' BCs/active='left right' Outputs/file_base=coupled_value_coupled_flux_penalty_out"
+      exodiff = 'coupled_value_coupled_flux_penalty_out.e'
+      prereq = 'interface_diffusion'
+      detail = 'using a hand-coded Jacobian'
+    []
+    [ad]
+      type = 'Exodiff'
+      input = 'ad-coupled-value-coupled-flux.i'
+      exodiff = 'ad-coupled-value-coupled-flux_out.e'
+      detail = 'using a Jacobian computed via automatic differentiation'
+      ad_indexing_type = 'global'
```

Comment:
Is it important to do the indexing globally here due to the interface itself?

Reply:
No this is because I use a variable functor in the test and the variable functor is only supported by global AD indexing. AD interface kernels alone work with both AD indexing types